pyccel/codegen/printing/ccode.py

```diff
@@ -1120,42 +1118,115 @@ def _print_Assign(self, expr):
         if isinstance(expr.rhs, FunctionCall) and isinstance(expr.rhs.dtype, NativeTuple):
             self._temporary_args = [VariableAddress(a) for a in expr.lhs]
             return '{};'.format(self._print(expr.rhs))
+        if isinstance(expr.rhs, (NumpyArray)):
+            return self.print_NumpyArray(expr)
+        if isinstance(expr.rhs, (NumpyFull)):
+            return self.print_NumpyFull(expr)
         lhs = self._print(expr.lhs)
-        rhs = expr.rhs
-        if isinstance(rhs, (NumpyArray)):
-            if rhs.rank == 0:
-                raise NotImplementedError(expr.lhs + "=" + expr.rhs)
-            dummy_array_name, _ = create_incremented_string(self._parser.used_names, prefix = 'array_dummy')
-            declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-            dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
+        rhs = self._print(expr.rhs)
+        return '{} = {};'.format(lhs, rhs)
+
+    def copy_NumpyArray_Data(self, expr):
+        """ print the Assignement of a NdArray
 
-            arg = rhs.arg
-            if rhs.rank > 1:
-                arg = functools.reduce(operator.concat, arg)
-            if isinstance(arg, Variable):
-                arg = self._print(arg)
+        parameters
+        ----------
+            expr : PyccelAstNode
+                The Assign Node used to get the lhs and rhs
+        Return
+        ------
+            String
+                Return a str that contain the declaration of a dummy data_buffer and
+                       a call to memcpy to copy it to NdArray struct
```

Comment:
Maybe you can be more general:
```
and a call to an operator which copies it to an NdArray struct
```
That way the docstring doesn't need updating when we fix issue #682